test/generators/StatusGenerator.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package generators
+
+import models.status._
+import org.scalacheck.Gen
+
+trait StatusGenerator extends BaseGenerator {
+
+  def rejectedStatusGen: Gen[SubmissionStatus] = Gen.oneOf(
+    Seq(DeRegistered,
+      SubmissionDecisionRejected,
+      SubmissionDecisionRevoked,
+      SubmissionDecisionExpired,
+      SubmissionWithdrawn,
+      DeRegistered
+    )
+  )
+
+  def activeStatusGen: Gen[SubmissionStatus] = Gen.oneOf(
+    Seq(
+      NotCompleted,
+      SubmissionReady,
+      SubmissionReadyForReview,
+      SubmissionDecisionApproved
```

Comment:
we should possibly have here renewal statuses

Reply:
it wont compile as ReadyForRenewal is derived from Renewal

Reply:
it will if you supply the renewal date, for example: ReadyForRenewal(Some(LocalDate.now()))
:)